WordPress/gutenberg
# Add visual regression tests for the Navigate Regions focus style

Related: https://github.com/WordPress/gutenberg/pull/43393

## What?

For now, visual regression testing is implemented only for a couple components in the Storybook and the tests only run locally. See #43393. However, there are several features in Gutenberg that would greatly benefit from visual regression tests.


## Why?

The Navigate Regions focus style is one of them. It already broke and was fixed a few times, for example in https://github.com/WordPress/gutenberg/pull/45369 and https://github.com/WordPress/gutenberg/pull/8554

Worth noting the focus style has been broken for quite a long time as it's hard to notice a breakage if the feature isn't actually used or manually tested. Normal E2E tests can test for the interaction to work as expected. Instead, only visual regression tests can test the blue outline focus style is actually **visible** and prevent from breaking again in the future. 

This PR is a proof of concept to consider what is the best way to introduce _more_ visual regression tests in the project.
Any thoughts and feedback welcome.
/Cc @kevin940726, @mirka, @aristath 

## How?

- Uses Playwright visual regression testing via `toHaveScreenshot()`.
- Screenshots target specific parts of the editor and not the full page, to reduce the reference images size.
- Adds 7 reference images in the `test/e2e/specs/editor/various/__snapshots__` directory, for a total size of ~57 KB.

Considerations:
- Maybe the visual regression tests should live in a dedicated directory `test/e2e/specs/visual-regression`.
- If the total repository size is a concern, this directory could maybe be git-ignored. Worth noting that when the reference images are missing, the related test will fail on a first run, but it will also generate the missing images. On the second run, the test will be able to use the new reference images.
- To my knowledge, the whole `test` directory is already excluded from the release build.
- When the test fails, 3 new images will be generated in the `artifacts/test-results` directory: 'expected', 'actual', and 'diff'. 
- The `toHaveScreenshot()` assertion has [a few optional parameters](https://playwright.dev/docs/test-assertions#locator-assertions-to-have-screenshot-2) that may come in handy,

## Testing Instructions




`npm run test:e2e:playwright -- editor/various/navigate-regions.spec.js`

To see the browser in action:
`npm run test:e2e:playwright -- --headed editor/various/navigate-regions.spec.js`

Only if you need to update the reference images  after edits to the test:
`$ npm run test:e2e:playwright -- --headed --update-snapshots editor/various/navigate-regions.spec.js`

### Testing Instructions for Keyboard


## Screenshots or screencast
